src/class-elements/private-field-as-async-arrow-function.case

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2019 Caio Lima. All rights reserved.
+// Copyright (C) 2018 Jaideep Bhoosreddy (Bloomberg LP). All rights reserved.
+// Copyright (C) 2018 Leo Balter. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+desc: Calling async arrow function returned from private field access
+info: |
+  Updated Productions
+
+  CallExpression[Yield, Await]:
+    CoverCallExpressionAndAsyncArrowHead[?Yield, ?Await]
+    SuperCall[?Yield, ?Await]
+    CallExpression[?Yield, ?Await]Arguments[?Yield, ?Await]
+    CallExpression[?Yield, ?Await][Expression[+In, ?Yield, ?Await]]
+    CallExpression[?Yield, ?Await].IdentifierName
+    CallExpression[?Yield, ?Await]TemplateLiteral[?Yield, ?Await]
+    CallExpression[?Yield, ?Await].PrivateName
+template: default
+features: [class-fields-private, async-functions, arrow-function]
+---*/
+
+//- elements
+#m = async () => 'test262';
+
+method() {
+  return this.#m();
+}
+//- assertions
+let c = new C();
+
+async function asyncRun() {
+    assert.sameValue(await c.method(), 'test262');
+}
+
+asyncRun();
```

Comment:
this won't really work without a few updates. Test262 tests need to indicate when tests are async so the engine will really wait for completion.

My suggestion is, add a flag for async in the frontmatter: `flags: [async]`

now you need to call `$DONE()` for completion, and give a "truthy" argument for failures, e.g.: `$DONE(1)`

For these calls, the Promise syntax should probably be easier to follow:

```js
c.method().then((value) => assert.sameValue(value, 'test262'))
  .then($DONE, $DONE);
```


Using the current async function:

```js
asyncRun().then($DONE, $DONE);
```

or maybe (definitely not the best candidate):

```js
async function asyncRun() {
  try {
    assert.sameValue(await c.method(), 'test262');
    $DONE();
  } catch(e) { $DONE(e); }
}

asyncRun();
```

Reply:
Thx for let me know this!